### ISSUE TITLE
NPE when sending message to trade partner (chat)

### ISSUE THREAD

**Issue by marko-lorentz** [bisq-network/bisq#5132]
### Description
NullPointerException on click of "Send" button in chat window (wanted to inform the trade partner that the transaction timed out and the trade was set to "invalid" state)

#### Version
v1.5.4

### Steps to reproduce
Send a message to trade partner of trade thvekt-9cb36979-cca7-4323-81fc-bf49619ac6a7-154 on my machine  produces the error every time, even after Bisq restart.

### Expected behaviour
Message is sent to trade partner

### Actual behaviour
Message not sent but error popup asking me to open this issue

### Screenshots
My input:
<img width="871" alt="image" src="https://user-images.githubusercontent.com/66749449/106355092-35431e00-62f6-11eb-9eea-3e77b394e65a.png">

Error: 
<img width="1001" alt="image" src="https://user-images.githubusercontent.com/66749449/106355060-0d53ba80-62f6-11eb-99b8-ebd326397d07.png">

#### Device or machine
MacBook Pro, Mac OS 11.1

#### Additional info
Stacktrace

```
Jan.-30 11:59:07.168 [JavaFX Application Thread] ERROR b.c.s.CommonSetup: Uncaught Exception from thread JavaFX Application Thread 
Jan.-30 11:59:07.170 [JavaFX Application Thread] ERROR b.c.s.CommonSetup: throwableMessage= null 
Jan.-30 11:59:07.170 [JavaFX Application Thread] ERROR b.c.s.CommonSetup: throwableClass= class java.lang.NullPointerException 
Jan.-30 11:59:07.172 [JavaFX Application Thread] ERROR b.c.s.CommonSetup: Stack trace:
java.lang.NullPointerException
	at bisq.desktop.main.shared.ChatView.lambda$sendDisputeDirectMessage$12(ChatView.java:651)
	at java.base/java.util.Optional.map(Optional.java:254)
	at bisq.desktop.main.shared.ChatView.sendDisputeDirectMessage(ChatView.java:647)
	at bisq.desktop.main.shared.ChatView.onSendMessage(ChatView.java:603)
	at bisq.desktop.main.shared.ChatView.onTrySendMessage(ChatView.java:526)
	at bisq.desktop.main.shared.ChatView.lambda$display$4(ChatView.java:215)
	at javafx.base/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at javafx.base/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
	at javafx.base/javafx.event.Event.fireEvent(Event.java:198)
	at javafx.graphics/javafx.scene.Node.fireEvent(Node.java:8865)
	at javafx.controls/javafx.scene.control.Button.fire(Button.java:200)
	at javafx.controls/com.sun.javafx.scene.control.behavior.ButtonBehavior.mouseReleased(ButtonBehavior.java:206)
	at javafx.controls/com.sun.javafx.scene.control.inputmap.InputMap.handle(InputMap.java:274)
	at javafx.base/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:218)
	at javafx.base/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at javafx.base/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.base/javafx.event.Event.fireEvent(Event.java:198)
	at javafx.graphics/javafx.scene.Scene$MouseHandler.process(Scene.java:3876)
	at javafx.graphics/javafx.scene.Scene$MouseHandler.access$1300(Scene.java:3604)
	at javafx.graphics/javafx.scene.Scene.processMouseEvent(Scene.java:1874)
	at javafx.graphics/javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2613)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:397)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:295)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(GlassViewEventHandler.java:434)
	at javafx.graphics/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:389)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:433)
	at javafx.graphics/com.sun.glass.ui.View.handleMouseEvent(View.java:556)
	at javafx.graphics/com.sun.glass.ui.View.notifyMouse(View.java:942)
	at javafx.graphics/com.sun.glass.ui.mac.MacView.notifyMouse(MacView.java:127)

```

**Comment by ghost** [comment]
This gives us enough information to reproduce the problem & fix the NPE.  Thanks for the high quality bug report!  :+1:

**Comment by ghost** [comment]
Two other examples of the same problem: 

- https://github.com/bisq-network/bisq/issues/3437
- https://github.com/bisq-network/bisq/issues/3325#issuecomment-538582764

**Comment by ghost** [comment]
The NPE is a symptom of a problem that occurred earlier during trade initialization process.  Either the trade or the trade contract is `null` so when sending a trader chat message it blows up when trying to access the trade contract.

@marko-lorentz, I would like to trace the events leading up to this.  Would there be any chance you could share a part of your log file from the time of the trade start until the NPE?  I'm curious to see if it is similar to #3437.

**Comment by marko-lorentz** [comment]
I produced a fresh log file (application startup, attempt to send the message to the chat, application shutdown). See: [Bisq log startup til shutdown.txt](https://github.com/bisq-network/bisq/files/5899409/Bisq.log.startup.til.shutdown.txt)

The trade with id `thvekt` itself _is_ broken. That's why I was trying to send a message to the trade partner. The log seems to indicate that the application loaded (at least a part of) the trade successfully. The application recognises the trade at 09:50:08.260 in the log. I think I switched to the transaction list in the UI to compose the message to the partner at 09:50:55.638. Crash follows at 09:51:36.412.


(The conditions that lead to the broken trade might be worth an extra examination as I think the trade partner will need a fee reimbursement and at least for me his entire offer can neither be accepted nor rejected any more - don't know whether the trade partner experiences a similar "deadlock" problem. I will try to find the reason that lead to the timeout error and failure in yesterdays log and open a dedicated issue.)


EDIT: 
Boring afternoon, so I cloned the git repository and had a short look at what happens.
`ChatView.sendDisputeDirectMessage:651` tries to call `supportSession.getClientPubKeyRing()` and therefore ends up in `TradeChatSession:54`. The trade object is available but the call to `trade.getContract()` returns `null` confirming your assumption that there's no contract.
